internal/domain/domain_operations.go

```diff
@@ -0,0 +1,242 @@
+		vcenterConfigRaw = *new([]interface{})
+		vcenterConfigRaw = append(vcenterConfigRaw, make(map[string]interface{}))
+	}
+	vcenterConfig := vcenterConfigRaw[0].(map[string]interface{})
+	vcenterConfig["id"] = domain.VCENTERS[0].ID
+	vcenterConfig["fqdn"] = domain.VCENTERS[0].Fqdn
+	_ = data.Set("vcenter", vcenterConfigRaw)
+
+	return domain, nil
+}
+
+func CreateDomainUpdateSpec(data *schema.ResourceData, markForDeletion bool) *models.DomainUpdateSpec {
+	result := new(models.DomainUpdateSpec)
+	if markForDeletion {
+		result.MarkForDeletion = true
+		return result
+	}
+	if data.HasChange("name") {
+		result.Name = data.Get("name").(string)
+	}
+
+	// TODO implement support for IPPoolSpecs in NsxTSpec
+	// by placing the added cluster spec in the DomainUpdateSpec
+	//nsxtSpec, err := generateNsxSpecFromResourceData(data)
+	//if err == nil {
+	//	result.NsxTSpec = nsxtSpec
+	//} else {
+	//	return nil, err
+	//}
+
+	return result
+}
+
+func ImportDomain(ctx context.Context, data *schema.ResourceData, apiClient *client.VcfClient,
+	domainId string, allowManagementDomain bool) ([]*schema.ResourceData, error) {
+	domainObj, err := SetBasicDomainAttributes(ctx, domainId, data, apiClient)
+	if err != nil {
+		return nil, err
+	}
+	if !allowManagementDomain && data.Get("type").(string) == "MANAGEMENT" {
+		return nil, fmt.Errorf("domain %s cannot be imported as it is management domain", domainId)
+	}
+
+	err = setClustersDataToDomainDataSource(domainObj.Clusters, ctx, data, apiClient)
+	if err != nil {
+		return nil, err
+	}
+	flattenedNsxClusterRef, err := network.FlattenNsxClusterRef(ctx, domainObj.NSXTCluster, apiClient)
+	if err != nil {
+		return nil, err
+	}
+	_ = data.Set("nsx_configuration", *flattenedNsxClusterRef)
+	return []*schema.ResourceData{data}, nil
+}
+
+func setClustersDataToDomainDataSource(domainClusterRefs []*models.ClusterReference, ctx context.Context, data *schema.ResourceData, apiClient *client.VcfClient) error {
+	clusterIds := make([]string, len(domainClusterRefs))
+	for i, clusterReference := range domainClusterRefs {
+		clusterIds[i] = *clusterReference.ID
+	}
+	// Sort the id slice, to have a deterministic order in every run of the domain datasource read
+	sort.Strings(clusterIds)
+
+	flattenedClusters := make([]map[string]interface{}, len(domainClusterRefs))
+	for i, clusterId := range clusterIds {
+		getClusterParams := clusters.GetClusterParams{ID: clusterId}
+		getClusterParams.WithContext(ctx).WithTimeout(constants.DefaultVcfApiCallTimeout)
+		clusterResult, err := apiClient.Clusters.GetCluster(&getClusterParams)
+		if err != nil {
+			return err
+		}
+		clusterRef := clusterResult.Payload
+		flattenedCluster, err := cluster.FlattenCluster(ctx, clusterRef, apiClient)
+		if err != nil {
+			return err
+		}
+		flattenedClusters[i] = *flattenedCluster
+
+	}
+	_ = data.Set("cluster", flattenedClusters)
+
+	return nil
+}
+
+func generateNsxSpecFromResourceData(data *schema.ResourceData) (*models.NsxTSpec, error) {
```

Comment:
generateNsxtSpec

Reply:
See my comment earlier. Where possible avoid NSX-T and use NSX.